### PR TITLE
Add tests for localhost-only node creation

### PR DESCRIPTION
**Pull request by Lobotuerk**
Tests added to up line-based coverage on rmw_implementations
Signed-off-by: lobotuerk <jtlorente@ekumenlabs.com>

### CODE REVIEW

File: test_rmw_implementation/test/test_create_destroy_node.cpp [hunk 1]
```diff
@@ -131,3 +131,41 @@ TEST_F(CLASSNAME(TestNodeConstructionDestruction, RMW_IMPLEMENTATION), create_an
   ASSERT_NE(nullptr, node) << rmw_get_error_string().str;
   EXPECT_EQ(RMW_RET_OK, rmw_destroy_node(node)) << rmw_get_error_string().str;
 }
+
+class CLASSNAME (TestNodeConstructionDestructionLocalHost, RMW_IMPLEMENTATION) : public ::testing::Test
```
Comment (hidmic) [inline test_rmw_implementation/test/test_create_destroy_node.cpp:135]: @Lobotuerk  nit:
```suggestion
class CLASSNAME (TestLocalhostNodeConstructionDestruction, RMW_IMPLEMENTATION) : public ::testing::Test
```

File: test_rmw_implementation/test/test_create_destroy_node.cpp [hunk 2]
```diff
@@ -131,3 +131,41 @@ TEST_F(CLASSNAME(TestNodeConstructionDestruction, RMW_IMPLEMENTATION), create_an
...
+    ret = rmw_init_options_fini(&options);
+    EXPECT_EQ(RMW_RET_OK, ret) << rcutils_get_error_string().str;
+  }
+
+  rmw_init_options_t options;
+  rmw_context_t context;
+};
+
+TEST_F(CLASSNAME(TestNodeConstructionDestructionLocalHost, RMW_IMPLEMENTATION), create_and_destroy) {
+  const char * const node_name = "my_node";
```
Comment (hidmic) [inline test_rmw_implementation/test/test_create_destroy_node.cpp:166]: @Lobotuerk nit: 
```suggestion
  constexpr char node_name[] = "/my_ns";
```

Same below.

**Review by hidmic — APPROVED** [review]
LGTM pending green CI

**Comment by hidmic** [comment]
> Add tests for non-localhost node creation #150

Isn't it `localhost-only node creation`?

File: test_rmw_implementation/test/test_create_destroy_node.cpp (hunk 2 above)
Comment (Lobotuerk) [reply test_rmw_implementation/test/test_create_destroy_node.cpp:166]: Does not seem to work like that, 
```
42: /home/lobo/rmw_tests/ros2_rolling/src/ros2/rmw_implementation/test_rmw_implementation/test/test_create_destroy_node.cpp:173: Failure
42: Expected: (nullptr) != (node), actual: (nullptr) vs NULL
42: invalid node name: node name must not contain characters other than alphanumerics or '_', at /home/lobo/rmw_tests/ros2_rolling/src/ros2/rmw_fastrtps/rmw_fastrtps_shared_cpp/src/rmw_node.cpp:58
```

File: test_rmw_implementation/test/test_create_destroy_node.cpp (hunk 1 above)
Comment (Lobotuerk) [reply test_rmw_implementation/test/test_create_destroy_node.cpp]: Updated on cdfa674

**Comment by Lobotuerk** [comment]
CI job up to `test_rmw_implementaiton`:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/12562/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/7530/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/10273/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/12484/)

File: test_rmw_implementation/test/test_create_destroy_node.cpp (hunk 2 above)
Comment (hidmic) [reply test_rmw_implementation/test/test_create_destroy_node.cpp:170]: Oof, I meant you to use `constexpr`, not the string ! My bad, it wasn't clear.

Comment (Lobotuerk) [reply test_rmw_implementation/test/test_create_destroy_node.cpp:170]: Should i change the other usages of `const char * const` on other parts of the file? I kept them for consistency

**Review by Blast545 — APPROVED** [review]
LGTM

**Comment by Lobotuerk** [comment]
Merging after green CI

**Pull request merged**